### PR TITLE
medusa-sanity-plugin

### PR DESCRIPTION
**Pull request by Aman-Kaushik13**

### CODE REVIEW

File: packages/medusa-plugin-contentful/src/services/__tests__/contentful.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import ContentfulService from "../contentful"
```
Comment (srindom) [inline packages/medusa-plugin-contentful/src/services/__tests__/contentful.js:1]: This shouldn't be changed

File: packages/medusa-plugin-sanity/__mocks__/contentful-management.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+"use strict";
```
Comment (srindom) [inline packages/medusa-plugin-sanity/__mocks__/contentful-management.js:1]: This file should be .gitignored

File: packages/medusa-plugin-sanity/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,10696 @@
+{
```
Comment (srindom) [inline packages/medusa-plugin-sanity/package-lock.json:1]: Please use yarn instead. i.e. delete this and include a yarn.lock instead

File: packages/medusa-plugin-sanity/src/api/routes/hooks/sanity.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+export default async (req, res) => {
+  try {
+    const sanityService = req.scope.resolve("sanityService")
+
+    //TODO: req sys in Sanity
+    const sanityType = req.body.sys.contentType.sys.id
+    const entryId = req.body.sys.id
```
Comment (srindom) [inline packages/medusa-plugin-sanity/src/api/routes/hooks/sanity.js:7]: Looks like sanity uses different ids than sys.id; 

Please make sure to update this and make sure that it works :)

File: packages/medusa-plugin-sanity/src/loaders/check-types.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+const checkContentTypes = async (container) => {
+  const contentfulService = container.resolve("contentfulService")
+
+  let product
+  let variant
+
+  try {
+    product = await contentfulService.getType("product")
+    variant = await contentfulService.getType("productVariant")
+  } catch (error) {
```
Comment (srindom) [inline packages/medusa-plugin-sanity/src/loaders/check-types.js:10]: `contentfulService` is not available here

**Review by srindom — CHANGES_REQUESTED** [review]
Hi! Thanks a lot for contributing :) 

I just went through a couple of the files in the PR and there still seems to be a lot of changes to be done. I think it would be really great if you could set up a test project like this one: https://github.com/medusajs/medusa-starter-contentful and get it up and running on your local machine :) Then you can link it up to a Sanity account and make sure that data is flowing correctly.

Let me know if you need some help with this!

**Comment by Aman-Kaushik13** [comment]
Hello thanks for reviewing. 
I had trouble with setting up tests which was related to **Jest**. I tried asking for help in the server, a thread opened but I didn't get the real solution and also I couldn't follow up regarding it later on.

**Comment by AshMarqz** [comment]
> _No description provided._
https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

**Comment by AshMarqz** [comment]
> Hi! Thanks a lot for contributing :)
> 
> I just went through a couple of the files in the PR and there still seems to be a lot of changes to be done. I think it would be really great if you could set up a test project like this one: https://github.com/medusajs/medusa-starter-contentful and get it up and running on your local machine :) Then you can link it up to a Sanity account and make sure that data is flowing correctly.
> 
> Let me know if you need some help with this!

https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-contentful/package-lock.json [hunk 6]
```diff
@@ -0,0 +1,8203 @@
+{
```
[the remark this answers is not in the document]
Comment (AshMarqz) [reply packages/medusa-plugin-contentful/package-lock.json:1]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-contentful/package.json [hunk 7]
```diff
@@ -27,8 +27,8 @@
   },
```
[the remark this answers is not in the document]
Comment (AshMarqz) [reply packages/medusa-plugin-contentful/package.json:27]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-contentful/src/services/__tests__/contentful.js (hunk 1 above)
Comment (AshMarqz) [reply packages/medusa-plugin-contentful/src/services/__tests__/contentful.js:1]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-sanity/__mocks__/contentful-management.js (hunk 2 above)
Comment (AshMarqz) [reply packages/medusa-plugin-sanity/__mocks__/contentful-management.js:1]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-sanity/package-lock.json (hunk 3 above)
Comment (AshMarqz) [reply packages/medusa-plugin-sanity/package-lock.json:1]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-sanity/src/api/routes/hooks/sanity.js (hunk 4 above)
Comment (AshMarqz) [reply packages/medusa-plugin-sanity/src/api/routes/hooks/sanity.js:7]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-sanity/src/loaders/check-types.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    if (!product) {
+      throw Error("Content type: `product` is missing in Contentful")
```
[the remark this answers is not in the document]
Comment (AshMarqz) [reply packages/medusa-plugin-sanity/src/loaders/check-types.js:12]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-sanity/src/loaders/check-types.js (hunk 5 above)
Comment (AshMarqz) [reply packages/medusa-plugin-sanity/src/loaders/check-types.js:10]: https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

**Comment by AshMarqz** [comment]
> Hello thanks for reviewing. I had trouble with setting up tests which was related to **Jest**. I tried asking for help in the server, a thread opened but I didn't get the real solution and also I couldn't follow up regarding it later on.

https://vuuzletvph.com/what-is-the-secret-of-nft-tokens-and-why-they-are-bought-for-millions-of-dollars/

Kindly check this out.Have a great day ahead!

File: packages/medusa-plugin-sanity/src/loaders/check-types.js (hunk 5 above)
Comment (vuuzle) [reply packages/medusa-plugin-sanity/src/loaders/check-types.js:10]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-sanity/src/loaders/check-types.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (vuuzle) [reply packages/medusa-plugin-sanity/src/loaders/check-types.js:12]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-sanity/src/api/routes/hooks/sanity.js (hunk 4 above)
Comment (vuuzle) [reply packages/medusa-plugin-sanity/src/api/routes/hooks/sanity.js:7]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-sanity/package-lock.json (hunk 3 above)
Comment (vuuzle) [reply packages/medusa-plugin-sanity/package-lock.json:1]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-sanity/__mocks__/contentful-management.js (hunk 2 above)
Comment (vuuzle) [reply packages/medusa-plugin-sanity/__mocks__/contentful-management.js:1]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-contentful/src/services/__tests__/contentful.js (hunk 1 above)
Comment (vuuzle) [reply packages/medusa-plugin-contentful/src/services/__tests__/contentful.js:1]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-contentful/package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (vuuzle) [reply packages/medusa-plugin-contentful/package.json:27]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-contentful/package-lock.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (vuuzle) [reply packages/medusa-plugin-contentful/package-lock.json:1]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

**Comment by riqwan** [comment]
Hey, thanks for the PR! Since v2 brought a lot of architectural and API changes on the backend, we will be closing this ticket.

**Pull request closed without merging**